File: springmvc-demo/src/main/java/com/gngsn/springmvc/controller/ApiController.java
package com.gngsn.springmvc.controller;

import com.gngsn.springmvc.dto.ResDTO;
import org.apache.shenyu.client.springmvc.annotation.ShenyuSpringMvcClient;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ApiController {

    @RequestMapping("/api")
    @ShenyuSpringMvcClient
    public ResDTO helloWorld() {
        return new ResDTO("[SpringMvc Demo] hello world");
    }

}

File: springmvc-demo-kt/src/main/java/com/gngsn/springmvc/controller/ApiController.kt
package com.gngsn.springmvc.controller

import com.gngsn.springmvc.ResDTO
import org.apache.shenyu.client.springmvc.annotation.ShenyuSpringMvcClient
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class ApiController {

    @RequestMapping("/kt/api")
    @ShenyuSpringMvcClient
    fun helloWorld(): ResDTO {
        return ResDTO("[SpringMvc Demo] hello world")
    }
}
